.github/actions/docker-setup/action.yaml

```diff
@@ -1,31 +1,35 @@
 name: "Docker build setup"
 description: |
   Runs an opinionated and unified docker build setup action. It does the following:
-  * Logs in to docker image registries (AWS ECR and GCP GAR)
+  * Logs in to docker image registries
+    * Only if the GCP args are provided will it log into GAR.
+    * Only if the AWS args are provided will it log into ECR.
   * Setup for buildx and other dependencies (crane)
   * Sets git credentials for private builds
 inputs:
   # GCP auth
   GCP_WORKLOAD_IDENTITY_PROVIDER:
-    required: true
+    required: false
```

Comment:
maybe can we default GAR on since it's our primary registry? if we ever want to push to ECR, we always push to GAR as well.

Reply:
I actually like this: we could have our private repository on GCP, and sync that to ECR somehow.

Reply:
Done, I made GAR required and ECR optional. Having something sync from GCP to ECR in the background sounds good, maybe easier to maintain than uploading to both.